cloudsync/tasks.py

```diff
@@ -165,23 +223,51 @@ def transcode_from_s3(self, video_id):  # pylint: disable=unused-argument
             video.update_status(VideoStatus.TRANSCODING)
 
 
-@shared_task(bind=True)
-def update_video_statuses(self):  # pylint: disable=unused-argument
+def process_watch_file(key):
     """
-    Check on statuses of all transcoding videos and update their status if appropriate
+    Copy the file from the watch bucket to the upload bucket, create model objects, and transcode
+
+    Args:
+        key(str): S3 object key
+
     """
-    transcoding_videos = Video.objects.filter(status=VideoStatus.TRANSCODING)
-    for video in transcoding_videos:
-        try:
-            refresh_status(video)
-        except EncodeJob.DoesNotExist:
-            # Log the exception but don't raise it so other videos can be checked.
-            log.exception("No EncodeJob object exists for video id %d", video.id)
-            video.update_status(VideoStatus.TRANSCODE_FAILED)
-        except ClientError as exc:
-            # Log the exception but don't raise it so other videos can be checked.
-            log.exception("AWS error when refreshing job status for video %d: %s", video.id, exc.response)
-            video.update_status(VideoStatus.TRANSCODE_FAILED)
+    watch_bucket = get_bucket(settings.VIDEO_S3_WATCH_BUCKET)
+
+    title = extract_title(key)
+    collection, _ = Collection.objects.get_or_create(title=title,
+                                                     owner=User.objects.get(username=settings.LECTURE_CAPTURE_USER))
```

Comment:
Yes, but I think just letting the exception raise here is ok? No other actions necessary because there is nothing yet that has to be modified, deleted, etc if it does happen?